### ISSUE TITLE
pypi deployment doesn't properly escape passwords

### ISSUE THREAD

**Issue by jrfonseca** [travis-ci/dpl#634]
I tried to use PyPI deployment for the first time.  But it failed

https://travis-ci.org/jrfonseca/xdot.py/builds/235166596

    /home/travis/.travis/job_stages: eval: line 54: unexpected EOF while looking for matching ``'
    /home/travis/.travis/job_stages: eval: line 55: syntax error: unexpected end of file

It took me a while to make sense of this, until I realized my password happened to have a

    `

character in it.

I've changed the password now, and it went further.

But this issue is bad at several levels:
- it prevents one using from using characters of their choice on their passwords
- it leaked information about characters on the password on to the log through bash's error message

**Comment by BanzaiMan** [comment]
Sorry for the troubles. We have a discussion on this subject in our document: https://docs.travis-ci.com/user/encryption-keys#Note-on-escaping-certain-symbols

**Comment by jrfonseca** [comment]
I don't think it's the same issue, as I didn't use `travis encrypt` but rather `travis setup pypi`, which asked me to interactively type the password in the terminal, which I did.

If indeed there was a problem when encrypting, then the bug would be in `travis setup pypi`, and not between the keyboard and the chair :)

But from the build log, I strongly suspect the problem is *not* when encrypting of the secret, but rather when using of the *decrypted* secret inside Travis virtual machine.  That is, I suspect the issue is in dpl, or whatever tool that wrote `/home/travis/.travis/job_stages` file...

**Comment by jrfonseca** [comment]
Hmm, https://github.com/travis-ci/travis.rb is for the travis CLI client, but like I explained, I believe the issue is in the travis setup/deployment scripts.

Anyway, I'm just a free tier user, and worked-around the issue on my side by changing password.  I've reported this mostly out of courtesy for great service I've been getting from Travis-CI, because I believe that interpreting backquotes in passwords with bash seems dangerous from a security standpoint (as it could easily lead to parts of passwords being output to the log in plaintext.)

But, if you think there's no security issue here (fingers crossed), only ease of use, then there's no action necessary as far as I'm concerned.
